### DISCUSSION TITLE
OAS 3 : allOf specification missing ?

### DISCUSSION THREAD

**Discussion by jonathan-vandebosch** [OAI/OpenAPI-Specification#2967] [Q&A]
Hello Everyone,

I was wondering why the allOf at array level with items was not documented, because it is working perfectly :

(in the screenshot the swagger has been simplified for clarity purposes)

![image](https://user-images.githubusercontent.com/65857453/177994486-726fac6c-03a5-45c2-8905-d01fa61f9ae3.png)

![image](https://user-images.githubusercontent.com/65857453/177994599-5b1063b5-3412-4642-a232-a384fa6e80b9.png)

But because of the spec mentionned at https://spec.openapis.org/oas/latest.html :

4.8.24.2.1 Composition and Inheritance (Polymorphism) [§](https://spec.openapis.org/oas/latest.html#composition-and-inheritance-polymorphism)
The OpenAPI Specification allows combining and extending model definitions using the allOf property of JSON Schema, in effect offering model composition. allOf takes an array of object definitions that are validated independently but together compose a single object.

While composition offers model extensibility, it does not imply a hierarchy between the models. To support polymorphism, the OpenAPI Specification adds the discriminator field. When used, the discriminator will be the name of the property that decides which schema definition validates the structure of the model. As such, the discriminator field MUST be a required field. There are two ways to define the value of a discriminator for an inheriting instance.

My company is using those specs to validate swaggers, and because of that my AllOf is refused.

**Comment by webron** [comment]
You're question is unclear. What exactly are you asking for/about?

**Reply by jonathan-vandebosch** [reply]
Hello Webron
According to the company I work for, the way that I'm using the keyword allOf is incorrect.
To validate my swagger, my company is using https://spec.openapis.org/oas/latest.html 
So this for my company is incorrect: 
> allOf:
>>   *_- $ref: '#/components/schemas/CardList'_*
>>   **_-  items_**

I don't agree with the decision of my company, they said that the second argument of allOf --> items doesn't fit with the spec definition located there : https://spec.openapis.org/oas/latest.html 

I need confirmation from you, that the way that I'm using allOf is correct or incorrect.

I joined screenshots above to show that everything was correctly interpreted

**Reply by webron** [reply]
This is more of a question about JSON Schema than OAS. Technically, your definition is valid. Whether it will do what you expect/want it to do depends on what that is.

**Reply by jonathan-vandebosch** [reply]
tks for the clarification Webron 👍

**Comment by handrews** [comment]
This was answered as far as it is relevant to OAS, with the remainder being directed towards JSON Schema.  Closing.
